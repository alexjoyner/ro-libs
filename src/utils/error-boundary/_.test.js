import React from 'react';
import { mount } from 'enzyme';
import { ErrorBoundary } from './index';

describe('`error-boundary`', () => {
  let component;
  beforeEach(() => {
    component = mount(<ErrorBoundary><h4>Hello</h4></ErrorBoundary>);
  });
  it('renders without exploding', () => {
    expect(component).toMatchSnapshot();
  });
  it('renders an h4 if there is no error', () => {
    expect(component.find('h4').exists()).toBe(true);
  });
  describe('When there is an error', () => {
    it('should render an h1 if there is an error', () => {
      component.setState({ hasError: true });
      expect(component.find('h1').exists()).toBe(true);
    });
  });
});
