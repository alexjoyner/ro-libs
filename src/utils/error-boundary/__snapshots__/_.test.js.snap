// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`error-boundary\` renders without exploding 1`] = `
<ErrorBoundary>
  <h4>
    Hello
  </h4>
</ErrorBoundary>
`;
